pyintegration/tests/test_parquet.py

```diff
@@ -0,0 +1,144 @@
+        col_inst1 = ColumnInstruction(column_name="y", parquet_column_name="py", codec_name="LZ4")
+
+        with self.subTest(msg="write_table(Table, str, max_dictionary_keys)"):
+            write_table(table, file_location, max_dictionary_keys=10)
+            self.assertTrue(os.path.exists(file_location))
+            shutil.rmtree(base_dir)
+        with self.subTest(msg="write_table(Table, str, col_instructions, max_dictionary_keys)"):
+            write_table(table, file_location, col_instructions=[col_inst, col_inst1], max_dictionary_keys=10)
+            self.assertTrue(os.path.exists(file_location))
+            shutil.rmtree(base_dir)
+        with self.subTest(msg="write_tables(Table[], destinations, col_definitions, "):
+            write_tables([table, table], [file_location, file_location2], col_definitions,
+                         col_instructions=[col_inst, col_inst1])
+            self.assertTrue(os.path.exists(file_location))
+            self.assertTrue(os.path.exists(file_location2))
+
+        # Reading
+        with self.subTest(msg="read_table(str)"):
+            table2 = read_table(path=file_location, col_instructions=[col_inst, col_inst1])
+
+        # Delete
+        with self.subTest(msg="delete(str)"):
+            if os.path.exists(file_location):
+                delete_table(file_location)
+                self.assertFalse(os.path.exists(file_location))
+            if os.path.exists(file_location2):
+                delete_table(file_location2)
+                self.assertFalse(os.path.exists(file_location2))
+        shutil.rmtree(base_dir)
+
+    def test_big_decimal(self):
+        j_type = dtypes.BigDecimal.j_type
+        big_decimal_list = [j_type.valueOf(301, 2),
+                            j_type.valueOf(201, 2),
+                            j_type.valueOf(101, 2)]
+        bd_col = InputColumn(name='decimal_value', data_type=dtypes.BigDecimal, input_data=big_decimal_list)
+        table = new_table([bd_col])
+        self.assertIsNotNone(table)
+        base_dir = os.path.join(self.root_dir, 'testCreation')
+        file_location = os.path.join(base_dir, 'table1.parquet')
+        if os.path.exists(file_location):
+            shutil.rmtree(file_location)
+
+        write_table(table, file_location)
+        table2 = read_table(file_location)
+        self.assertEqual(table.size, table2.size)
+        self.assertEqual(table, table2)
+
+        self.assertTrue(os.path.exists(file_location))
+        shutil.rmtree(base_dir)
+
+    @classmethod
+    def tearDownClass(cls):
+        # remove the junk definitions created in the tests, if they exist...
+        if os.path.exists(cls.root_dir):
+            try:
+                shutil.rmtree(cls.root_dir)
+            except Exception as e:
+                print("Tried removing directory {}, but failed with error {}. "
+                      "Manual clean-up may be necessary".format(cls.root_dir, e))
```

Comment:
I would put this at the top by the setup method, so that it is very clear what is happening.  Until I got down here, I assumed there was no teardown.